src/status_im/common/biometric/events.cljs

```diff
@@ -52,81 +45,32 @@
   {:events [:biometric/show-message]}
   [_ code]
-  (let [handle-error? (and code
-                           (not (contains? #{constants/biometric-error-user-canceled
-                                             constants/biometric-error-user-fallback}
-                                           code)))
-        content       (if (#{constants/biometric-error-not-available
-                             constants/biometric-error-not-enrolled}
-                           code)
-                        (i18n/label :t/grant-face-id-permissions)
-                        (i18n/label :t/biometric-auth-error {:code code}))]
-    (when handle-error?
-      {:effects.utils/show-popup
-       {:title   (i18n/label :t/biometric-auth-login-error-title)
-        :content content}})))
-
-(defn- supress-biometry-error-key
-  [key-uid]
-  (keyword (str "biometric/supress-not-enrolled-error-" key-uid)))
-
-;; NOTE: if the account had biometrics registered, but it's not enrolled at the moment,
-;; we should show the error message only once and supress further "NOT_ENROLLED" errors
-;; until biometry is enrolled again. Note that we can only know that when :biometric/authenticate
-;; is dispatched and fails with "NOT_ENROLLED", since :biometric/get-supported-biometric-type
-;; only tells us what kind of biometric is available on the device, but it doesn't know of its
-;; enrollment status.
-(re-frame/reg-fx
- :biometric/supress-not-enrolled-error
- (fn [[key-uid dispatch-event]]
-   (let [storage-key (supress-biometry-error-key key-uid)]
-     (-> (async-storage/get-item storage-key identity)
-         (.then (fn [item]
-                  (when (not item)
-                    (rf/dispatch dispatch-event)
-                    (async-storage/set-item! storage-key true))))
-         (.catch (fn [err]
-                   (log/error "Couldn't supress biometry NOT_ENROLLED error"
-                              {:key-uid key-uid
-                               :event   :biometric/supress-not-enrolled-error
-                               :error   err})))))))
+  (let [content (if (#{biometrics/biometric-error-not-available
+                       biometrics/biometric-error-not-enrolled}
+                     code)
+                  (i18n/label :t/grant-face-id-permissions)
+                  (i18n/label :t/biometric-auth-error {:code code}))]
+    {:effects.utils/show-popup
+     {:title   (i18n/label :t/biometric-auth-login-error-title)
+      :content content}}))
 
-;; NOTE: when biometrics is re-enrolled, we erase the flag in async-storage to assure
-;; the "NOT_ENROLLED" error message will be shown again if biometrics is un-enrolled
-;; in the future.
-(re-frame/reg-fx
- :biometric/reset-not-enrolled-error
- (fn [key-uid]
-   (let [storage-key (supress-biometry-error-key key-uid)]
-     (-> (async-storage/get-item storage-key identity)
-         (.then (fn [supress?]
-                  (when supress?
-                    (async-storage/set-item! storage-key nil))))
```

Comment:
@clauxx : are you sure we don't need this flow anymore?

Reply:
I mean using `async-storage` to remember user's decision on whether to allow biometric access or not, this decision/setting persists across multiple accounts on the same device ( if the user has more than 1 account on a device )

Reply:
This was doing something else. Here we're storing a flag for suppressing the "not enrolled" error, which would appear when the user disabled biometrics on the device. The error would appear every time when starting the app with a profile that has biometrics enabled (during registration) while device biometrics was disabled. Instead, now we're checking if biometrics is enabled before running biometric authentication, which removes the need for all this suppression business.

Reply:
Ah sounds good, thanks for explaining.